### DISCUSSION TITLE
Use variables in required scripts

### DISCUSSION THREAD

**Discussion by Sebaestschjin** [AnWeber/vscode-httpyac#103] [Q&A] [answered]
Hi.

Is it possible to access the available variables within required scripts?

My current goal is to define some default headers for all request in a file. Some of the header values come from the environment, some others from imported variables. I can achieve that with the `+request` hook and manipulate the headers there. Access to all variables is also possible there. However, since there are multiple headers to add and I also want to use that same function in multiple files, I extracted the functionality into a separate script and simply `require` that in all files at the top. But then I don't have access to the current variables. I could of course pass all the values I need as a parameter to the function in my script, but that somehow defeats the purpose of the script I think and makes it harder to update.
Any suggestions for that?

E.g. my minimal test example is this:

Request:
```
{{+request
    require("./script").setHeaders(request)
}}

###
GET http://example.org
```

Script:
```js
module.exports = {
  setHeaders: (request) => {
    request.headers["SomeHeader"] = some_value;
    request.headers["SomeOtherHeader"] = some_other_value;
  },
};
```

**Comment by AnWeber** [comment, marked as answer]
The easiest thing to do is to pass you the variables as objects. Either just use `this` or in my eyes better, use the variables explicitly.

```
{{+request
    require("./script").setHeaders(request, this)
}}
```

```
{{+request
    require("./script").setHeaders(request, { some_value, some_other_value})
}}
```

If you require the script in several places, you can have a look at the [hook api](https://github.com/httpyac/httpyac.github.io/tree/main/examples/plugins/requestHook). That's exactly what it's for and it has easier access to the variables and interface. With this I extend httpyac with the companies internal api

**Reply by Sebaestschjin** [reply]
Thanks a lot for the very fast answer!
I think's I'll start with the `this` approach for now, to get that starting quickly, and will also take a look at the hook api. That's the part I didn't read, yet in the documentation...
